examples/twittermap/web/public/javascripts/map/controllers.js

```diff
@@ -683,23 +690,56 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common','clou
         // Caching feature only works when the given threshold is greater than zero.
         if (cloudberryConfig.cacheThreshold > 0) {
             Cache.getCityPolygonsFromCache(rteBounds).done(function(data) {
-                $scope.geojsonData.city = data;
 
-                if($scope.polygons.cityPolygons) {
-                    $scope.map.removeLayer($scope.polygons.cityPolygons);
+                if($scope.addCityLayerRequest === true){
+                    $scope.geojsonData.city = data;
+                    $scope.polygons.cityPolygons = L.geoJson(data, {
+                      style: $scope.styles.cityStyle,
+                      onEachFeature: onEachFeature
+                  });
                 }
-                $scope.polygons.cityPolygons = L.geoJson(data, {
-                    style: $scope.styles.cityStyle,
-                    onEachFeature: onEachFeature
-                });
+
                 //set center and boundary done by Cache
                 if (!$scope.status.init) {
-                    resetGeoIds($scope.bounds, $scope.geojsonData.city, 'cityID');
+                    resetGeoIds($scope.bounds, data, 'cityID');
                     cloudberry.parameters.geoLevel = 'city';
                     cloudberry.query(cloudberry.parameters);
                 }
                 resetGeoInfo("city");
-                $scope.map.addLayer($scope.polygons.cityPolygons);
+
+                // incrementally add polygons after user zoom in
+                if($scope.addCityLayerRequest === false){
+                    var diffData = [];
+
+                    // finding out the new incoming cities which are not stored in geojsonData.city yet
+                    for(i =0; i < data.features.length; i++){
+                        var overlap = false;
+
+                        for(j =0; j < $scope.geojsonData.city.features.length; j++){
```

Comment:
Can we make a set that includes "$scope.geojsonData.city.features[j].properties.cityID" and use it like putValues() in mapResultCache?